spec/unit/daemon_spec.rb

```diff
@@ -45,15 +45,24 @@ def run_loop(jobs)
   end
 
   describe "when setting signal traps" do
-    signals = {:INT => :stop, :TERM => :stop }
-    signals.update({:HUP => :restart, :USR1 => :reload, :USR2 => :reopen_logs}) unless Puppet.features.microsoft_windows?
-    signals.each do |signal, method|
-      it "should log and call #{method} when it receives #{signal}" do
-        Signal.expects(:trap).with(signal).yields
+    [:INT, :TERM].each do |signal|
+      it "logs a notice and exits when sent #{signal}" do
+        Signal.stubs(:trap).with(signal).yields
```

Comment:
This is a slightly evil way to test the code. Normally `Signal.trap` just registers a signal handler to be invoked when it receives the signal at some time in the future. But this stub causes the handler to be invoked when `set_signal_traps` is called. This pattern existed previously (above on line 52), so I left it as is.